### ISSUE TITLE
ACK for MemoryDB Subnet Issue

### ISSUE THREAD

**Issue by evdevr** [aws-samples/rancher-on-aws-workshop#65]
getting an error with MemoryDB cluster create. looks like not all AZs support MemoryDB.

```
kubectl apply -f memorydb-subnetgroup.yaml ; kubectl describe subnetgroup "${MEMORYDB_SUBNETGROUP_NAME}"

subnetgroup.memorydb.services.k8s.aws/subnet-workshop-memorydb-cluster unchanged
Name:         subnet-workshop-memorydb-cluster
Namespace:    default
Labels:       <none>
Annotations:  <none>
API Version:  memorydb.services.k8s.aws/v1alpha1
Kind:         SubnetGroup
Metadata:
  Creation Timestamp:  2023-09-05T11:54:54Z
  Finalizers:
    finalizers.memorydb.services.k8s.aws/SubnetGroup
  Generation:  1
  Managed Fields:
    API Version:  memorydb.services.k8s.aws/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .:
          v:"finalizers.memorydb.services.k8s.aws/SubnetGroup":
    Manager:      controller
    Operation:    Update
    Time:         2023-09-05T11:54:54Z
    API Version:  memorydb.services.k8s.aws/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:description:
        f:name:
        f:subnetIDs:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2023-09-05T11:54:54Z
    API Version:  memorydb.services.k8s.aws/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:ackResourceMetadata:
          .:
          f:ownerAccountID:
          f:region:
        f:conditions:
    Manager:         controller
    Operation:       Update
    Subresource:     status
    Time:            2023-09-05T11:55:01Z
  Resource Version:  15897
  UID:               6f377d2b-3ec3-44cd-bc28-0e233baf10d7
Spec:
  Description:  MemoryDB cluster subnet group
  Name:         subnet-workshop-memorydb-cluster
  Subnet I Ds:
    subnet-071becc2bda9fd0c4
    subnet-0e0a9b35807b9cd62
    subnet-0fb5eac5fbf0ba200
    subnet-08ae10c4d17c30095
    subnet-029e22d1872aa98ab
    subnet-0b54597b9b3a2ef6c
Status:
  Ack Resource Metadata:
    Owner Account ID:  184917398746
    Region:            us-east-1
  Conditions:
    Message:               SubnetNotAllowedFault: Subnets: [subnet-071becc2bda9fd0c4, subnet-08ae10c4d17c30095] are not in a supported availability zone. Supported availability zones are [us-east-1d, us-east-1a, us-east-1c].
    Status:                True
    Type:                  ACK.Terminal
    Last Transition Time:  2023-09-05T11:55:01Z
    Message:               Resource not synced
    Reason:                resource is in terminal condition
    Status:                False
    Type:                  ACK.ResourceSynced
Events:                    <none>
```

**Comment by marshkkm** [comment]
This needs the filter to be updated to not allow the those AZs. Fixing now.

Will need to be updated when support for other regions is added

**Comment by zackbradys** [comment]
updated and fixed in PR #66
